lib/processors/manifestEnhancer.js

```diff
@@ -0,0 +1,417 @@
+			// Relative and server absolute URLs are reserved for future use.
+			return null;
+		}
+
+		sNoScheme = sNoScheme.replace("//", "");
+
+		return "/resources/" + sNoScheme;
+	} else {
+		// not a ui5 url
+		return sUrl;
+	}
+}
+
+/**
+ * Normalizes a bundle URL relative to the project namespace.
+ *
+ * @param {string} bundleUrl Relative bundle URL to be normalized
+ * @param {string} sapAppId Project namespace from sap.app/id (e.g. "sap.ui.demo.app")
+ *                          to which the URL is relative to
+ * @returns {string} Normalized relative bundle URL (e.g. "i18n/i18n.properties")
+ */
+function normalizeBundleUrl(bundleUrl, sapAppId) {
+	// Create absolute path with namespace from sap.app/id
+	const absoluteNamespace = `/resources/${sapAppId.replaceAll(/\./g, "/")}`;
+
+	const resolvedAbsolutePath = path.resolve(absoluteNamespace, bundleUrl);
+	const resolvedRelativePath = path.relative(absoluteNamespace, resolvedAbsolutePath);
+	return resolvedRelativePath;
+}
+
+/**
+ * Returns the bundle URL from the given bundle configuration.
+ *
+ * @param {object} bundleConfig Bundle configuration
+ * @param {string} sapAppId Project namespace from sap.app/id (e.g. "sap.ui.demo.app")
+ *                          to which a bundleName should be resolved to
+ * @param {string} [defaultBundleUrl] Default bundle url in case bundleConfig is not defined
+ */
+function getBundleUrlFromConfig(bundleConfig, sapAppId, defaultBundleUrl) {
+	if (!bundleConfig) {
+		// Use default URL (or undefined if argument is not provided)
+		return defaultBundleUrl;
+	} else if (typeof bundleConfig === "string") {
+		return bundleConfig;
+	} else if (typeof bundleConfig === "object") {
+		return getBundleUrlFromConfigObject(bundleConfig, sapAppId);
+	}
+}
+
+// Same as above, but does only accept objects, not strings or defaults
+function getBundleUrlFromConfigObject(bundleConfig, sapAppId, fallbackBundleUrl) {
+	if (typeof bundleConfig === "object") {
+		if (bundleConfig.bundleName) {
+			return getRelativeBundleUrlFromName(bundleConfig.bundleName, sapAppId);
+		} else if (bundleConfig.bundleUrl) {
+			return bundleConfig.bundleUrl;
+		}
+	}
+	return fallbackBundleUrl;
+}
+
+// See runtime logic in sap/ui/core/Lib#_normalizeI18nSettings
+function getBundleUrlFromSapUi5LibraryI18n(vI18n) {
+	if (vI18n == null || vI18n === true) {
+		return "messagebundle.properties";
+	} else if (typeof vI18n === "string") {
+		return vI18n;
+	} else if (typeof vI18n === "object") {
+		return vI18n.bundleUrl;
+	} else {
+		return null;
+	}
+}
+
+class ManifestEnhancer {
+	/**
+	 * @param {string} manifest manifest.json content
+	 * @param {string} filePath manifest.json file path
+	 * @param {fs} fs Node fs or custom [fs interface]{@link module:@ui5/fs/fsInterface}
+	 */
+	constructor(manifest, filePath, fs) {
+		this.fsReadDir = promisify(fs.readdir);
+		this.cwd = path.dirname(filePath);
+		this.manifest = JSON.parse(manifest);
```

Comment:
Add filePath so we can use it in logging (see below)

```suggestion
		this.filePath = filePath;
		this.manifest = JSON.parse(manifest);
```